### DISCUSSION TITLE
Desktop Notifications

### DISCUSSION THREAD

**Discussion by Findus23** [Hypfer/Valetudo#1303] [Archive]
One feature that would be quite nice (and potentially not too complex to implement) would be desktop notifications for events (like finishing a room cleanup).

Of course having these work in the background is quite complex, so I think having them just work as long as the website is open should make this a lot more doable.

**Comment by Hypfer** [comment]
Unfortunately, these APIs require you to use HTTPs which is not feasible for devices without some kind of (in theory) public domain address due to how that whole certificate stuff works currently.

See https://developer.mozilla.org/en-US/docs/Web/API/notification

To fix this, browser vendors either need to whitelist private IP ranges for non-https access to these apis or there needs to be a way to allow https certificates without requiring some central authority to sign your stuff.

As it is now, self-signed certificates unfortunately are just painful to use. 
You might as well just suggest installing gentoo from a stage 2 tarball to your mom because her printer doesn't work.

**Reply by Findus23** [reply]
Ah, thanks for the info. I totally missed that those APIs are HTTPS-only and I agree that making Valetudo HTTPS in a non-hackish way seems impossible at the moment.

**Reply by Hypfer** [reply]
There's this Firefox Bugzilla Bug about the issue:
https://bugzilla.mozilla.org/show_bug.cgi?id=1705543

Furthermore, there's this:
https://github.com/httpslocal/proposals
https://httpslocal.github.io/proposals/

There's also this essay about the issue from 2018:
https://dev.to/danielkun/where-is-https-for-iot-49ao

Overall this is just immensely frustrating

**Reply by Hypfer** [reply]
btw this also blocks installing valetudo as a PWA :(
